### ISSUE TITLE
Decimal backed FloatFields

### ISSUE THREAD

**Issue by nwmartin** [goldcaddy77/warthog#239]
Right now the FloatField warthog decorator for postgres will actually default to `float8`. See:https://github.com/goldcaddy77/warthog/blob/master/src/decorators/FloatField.tshttps://github.com/goldcaddy77/warthog/blob/master/src/torm/data-types.ts

There's basically two approaches here: we can make a new decimal field decorator or we can update FloatField to allow manually overriding the backing type. Either works. We really want this for storing monetary values even if the in-memory type is going to be a garbage javascript float. We'll interact with the database values through other means and do calculations and the like as integers.

**Comment by goldcaddy77** [comment]
Did a ton of work on data types today.  https://github.com/goldcaddy77/warthog/pull/250/files

New decorator: https://github.com/goldcaddy77/warthog/pull/250/files#diff-66d28b7a7fff0ea0faa35f74adf996c8R1

Usage: https://github.com/goldcaddy77/warthog/pull/250/files#diff-984f57b34defd9f4be76f9d2e7979a8cR128-R129

I still need to add the ability to pass precision in, but it's close.

**Comment by goldcaddy77** [comment]
All set:  https://github.com/goldcaddy77/warthog/pull/250/files#diff-984f57b34defd9f4be76f9d2e7979a8cR89-R93.  Looks like this:

```typescript
  @NumericField({ nullable: true })
  numericField?: number;

  @NumericField({ nullable: true, precision: 5, scale: 2 })
  numericFieldCustomPrecisionScale?: number;
```

**Comment by goldcaddy77** [comment]
Published in version 2.0.0-beta.8
